node/Dockerfile

```diff
@@ -0,0 +1,64 @@
+# Miden node Dockerfile
+
+# Setup image builder
+FROM rust:1.76-slim-bookworm AS builder
+
+# Set labels
+LABEL org.opencontainers.image.authors=miden@polygon.io \
+      org.opencontainers.image.url=https://0xpolygonmiden.github.io/ \
+      org.opencontainers.image.documentation=https://github.com/0xPolygonMiden/miden-node \
+      org.opencontainers.image.source=https://github.com/0xPolygonMiden/miden-node \
+      org.opencontainers.image.vendor=Polygon \
+      org.opencontainers.image.licenses=MIT
+
+ARG CREATED
+ARG VERSION
+ARG COMMIT
+LABEL org.opencontainers.image.created=$CREATED \
+      org.opencontainers.image.version=$VERSION \
+      org.opencontainers.image.revision=$COMMIT
+
+# Install dependencies
+RUN apt-get update && \
+    apt-get -y upgrade && \
+    apt-get install -y llvm clang bindgen pkg-config libssl-dev libsqlite3-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+# Copy source code
+WORKDIR /app
+COPY . .
+
+# Cache Cargo dependencies
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    cargo fetch --manifest-path node/Cargo.toml
+
+# Build the node crate
+RUN cargo install --features testing --path node
+RUN miden-node make-genesis --inputs-path node/genesis.toml
+
+# Run Miden node
+FROM debian:bookworm-slim
```

Comment:
Note: The final image has an okay size:

```
REPOSITORY                  TAG                 IMAGE ID      CREATED             SIZE
localhost/miden-node-image  latest              63b9223ac3cb  About a minute ago  124 MB
```

Reply:
Note: `start`/`stop` is working, the state is not lost

Reply:
> Note: The final image has an okay size:
> 
> ```
> REPOSITORY                  TAG                 IMAGE ID      CREATED             SIZE
> localhost/miden-node-image  latest              63b9223ac3cb  About a minute ago  124 MB
> ```

Do you have ideas in mind to improve on it?

Reply:
> Note: `start`/`stop` is working, the state is not lost

This is positive right?

Reply:
> Do you have ideas in mind to improve on it?

I think this size is good enough. I was just adding context to the PR.

> Note: start/stop is working, the state is not lost

Yes. This was not a request. I was just adding context to the PR.